vector<vector<bool>> grid
---
vector<vector<bool>> cleanedGrid